src/mds/Server.cc

```diff
@@ -6253,6 +6261,57 @@ void Server::handle_client_removexattr(MDRequestRef& mdr)
   journal_and_reply(mdr, cur, 0, le, new C_MDS_inode_update_finish(this, mdr, cur));
 }
 
+void Server::handle_client_getvxattr(MDRequestRef& mdr)
+{
+  const cref_t<MClientRequest> &req = mdr->client_request;
+  string name(req->get_path2());
+
+  // is a ceph virtual xattr?
+  if (!is_ceph_vxattr(name)) {
+    respond_to_request(mdr, -CEPHFS_EINVAL);
+    return;
+  }
+
+  // can't use rdlock_path_pin_ref because we need to xlock snaplock/policylock
+  CInode *cur = try_get_auth_inode(mdr, req->get_filepath().get_ino());
+  if (!cur) {
+    return;
+  }
+  if (!cur->is_dir()) {
+    respond_to_request(mdr, -CEPHFS_ENOTDIR);
+    return;
+  }
+
+  // handle these vxattrs
+  int r = 0;
+  ceph::bufferlist bl;
+  if (name == "ceph.dir.layout"sv) {
```

Comment:
I think you should only support getting `ceph.dir.layout` and its value will depend on whether it is inherited. See how `handle_client_openc` looks up the inherited layout.

And, in fact, the types of locks you acquire will depend on the vxattr being looked up. Look at `handle_set_vxattr`.

Reply:
what about the existing support for getting individual layout fields ?
does that support remain the old way ?